backend/server.js

```diff
@@ -13,15 +16,178 @@ const port = process.env.PORT || 8080;
 const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors());
+app.use(
+  cors({
+    origin: "https://emilia-michelle-project-auth.netlify.app", 
+    methods: ["GET", "POST"], // Specify the allowed methods
+    allowedHeaders: ["Content-Type", "Authorization"], // Specify the allowed headers
+  })
+); 
 app.use(express.json());
 
+// Defines endpoint paths as constants to be able to only update the paths in one place if needed
+const PATHS = {
+  root: "/",
+  register: "/register",
+  login: "/login",
+  secrets: "/secrets",
+  sessions: "/sessions"
+}
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send(listEndpoints(app));
+});
+
+const { Schema } = mongoose;
+
+const UserSchema = new mongoose.Schema({
+  name: {
+    type: String,
+    required: true,
+    unique: true
+  },
+  password: {
+    type: String,
+    required: true
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString("hex")
+  }
+});
+
+const User = mongoose.model("User", UserSchema);
+
+// Registration
+app.post(PATHS.register, async (req, res) => {
+  const { name, password } = req.body;
+
+  try {
+    const salt = bcrypt.genSaltSync();
+    // Do not store plaintext passwords
+    const newUser = await new User({
+      name: name,
+      password: bcrypt.hashSync(password, salt)})
+    .save();
+    res.status(201).json({
+      success: true,
+      response: {
+        name: newUser.name,
+        id: newUser._id,
+        accessToken: newUser.accessToken
+      }
+    })
+  } catch (e) {
+    // Bad request
+    res.status(400).json({
```

Comment:
We like how you solved the error-messages in your code - clear and thorough all the way.